Content.Server/Chemistry/EntitySystems/SolutionPurgeSystem.cs

```diff
@@ -0,0 +1,60 @@
+using Content.Server.Chemistry.Components;
+using Content.Server.Chemistry.Components.SolutionManager;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Chemistry.EntitySystems;
+
+public sealed class SolutionPurgeSystem : EntitySystem
+{
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SolutionPurgeComponent, EntityUnpausedEvent>(OnUnpaused);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SolutionPurgeComponent, SolutionContainerManagerComponent>();
+        while (query.MoveNext(out var uid, out var purge, out var manager))
+        {
+            if (_timing.CurTime < purge.NextPurgeTime)
+                continue;
+
+            // timer ignores if it's empty, it's just a fixed cycle
+            purge.NextPurgeTime = _timing.CurTime + purge.Duration;
```

Comment:
```suggestion
            // timer ignores if it's empty, it's just a fixed cycle
            purge.NextPurgeTime += purge.Duration;
```

Otherwise you get rounding issues over time.

Reply:
Fixed in a2bffd7